supporter-product-data/src/main/scala/com/gu/lambdas/FetchResultsLambda.scala

```diff
@@ -0,0 +1,44 @@
+package com.gu.lambdas
+
+import com.amazonaws.services.lambda.runtime.Context
+import com.gu.conf.ZuoraQuerierConfig
+import com.gu.lambdas.FetchResultsLambda.fetchResults
+import com.gu.model.Stage
+import com.gu.model.states.{FetchResultsState, UpdateDynamoState}
+import com.gu.model.zuora.response.JobStatus.Completed
+import com.gu.monitoring.SafeLogger
+import com.gu.okhttp.RequestRunners.configurableFutureRunner
+import com.gu.services.{S3Service, SelectActiveRatePlansQuery, ZuoraQuerierService}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.DurationInt
+
+class FetchResultsLambda extends Handler[FetchResultsState, UpdateDynamoState] {
+  override protected def handlerFuture(input: FetchResultsState, context: Context) =
+    fetchResults(Stage.fromEnvironment, input.jobId)
+}
+
+object FetchResultsLambda {
+  def fetchResults(stage: Stage, jobId: String) = {
+    SafeLogger.info(s"Attempting to fetch results for jobId $jobId")
+    for {
+      config <- ZuoraQuerierConfig.load(stage)
+      service = new ZuoraQuerierService(config, configurableFutureRunner(60.seconds))
+      result <- service.getResults(jobId)
+      _ = assert(result.status == Completed, s"Job with id $jobId is still in status ${result.status}")
+      batch = result.batches.headOption.toRight(s"No batches were returned in the batch query response for jobId $jobId").right.get
+      fileId = batch.fileId.toRight(s"Batch.fileId was missing in jobId $jobId").right.get
```

Comment:
doing right.get on an either doesn't actually report the contents of the left in the exception - that only works for `Try`.
```scala
    def get: B = e match {
      case Right(b) => b
      case _        => throw new NoSuchElementException("Either.right.get on Left")
    }
```

Reply:
That's annoying...